python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -906,6 +906,107 @@ def test_pandas_datetime_to_date64_failures(self, mask):
         with pytest.raises(pa.ArrowInvalid, match=expected_msg):
             pa.Array.from_pandas(s, type=pa.date64(), mask=mask)
 
+    def test_array_date_as_object(self):
+        data = [date(2000, 1, 1),
+                None,
+                date(1970, 1, 1),
+                date(2040, 2, 26)]
+        expected = np.array(['2000-01-01',
+                             None,
+                             '1970-01-01',
+                             '2040-02-26'], dtype='datetime64')
+
+        arr = pa.array(data)
+        assert arr.equals(pa.array(expected))
+
+        result = arr.to_pandas()
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(arr.to_pandas(), expected)
+
+        result = arr.to_pandas(date_as_object=True)
+        expected = expected.astype(object)
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(result, expected)
+
+    def test_array_date_as_object(self):
+        data = [date(2000, 1, 1),
+                None,
+                date(1970, 1, 1),
+                date(2040, 2, 26)]
+        expected = np.array(['2000-01-01',
+                             None,
+                             '1970-01-01',
+                             '2040-02-26'], dtype='datetime64')
+
+        arr = pa.array(data)
+        assert arr.equals(pa.array(expected))
+
+        result = arr.to_pandas()
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(arr.to_pandas(), expected)
+
+        result = arr.to_pandas(date_as_object=True)
+        expected = expected.astype(object)
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(result, expected)
+
+    def test_chunked_array_date_as_object(self):
+        data = [date(2000, 1, 1),
+                None,
+                date(1970, 1, 1),
+                date(2040, 2, 26)]
+        expected = np.array(['2000-01-01',
+                             None,
+                             '1970-01-01',
+                             '2040-02-26'], dtype='datetime64')
+        carr = pa.chunked_array([data])
+
+        result = carr.to_pandas()
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(carr.to_pandas(), expected)
+
+        result = carr.to_pandas(date_as_object=True)
+        expected = expected.astype(object)
+        assert result.dtype == expected.dtype
+        npt.assert_array_equal(result, expected)
+
+    def test_column_date_as_object(self):
+        data = [date(2000, 1, 1),
+                None,
+                date(1970, 1, 1),
+                date(2040, 2, 26)]
+        expected = np.array(['2000-01-01',
+                             None,
+                             '1970-01-01',
+                             '2040-02-26'], dtype='datetime64')
+
+        arr = pa.array(data)
+        column = pa.column('date', arr)
+
+        result = column.to_pandas()
```

Comment:
`Column.to_pandas` returns a `pd.Series` whereas `Array.to_pandas()` and `ChunkedArray.to_pandas()` returns `ndarrays`. Is it a feature or a bug? Either way it feels weird and inconsistent, I wouldn't expect ndarray as the result of `to_pandas()`.

Reply:
So "to_pandas" is returning a data representation suitable for pandas (i.e. with the right null special values, type casts for integers, etc.) so neither is wrong, per se, but it would good to be consistent

Reply:
As far as I understood is a `Series` a _named_ array with an index. We never have an index but the `Column` object has a name.

Reply:
Verdict?

Reply:
I think we should just return an ndarray from Column.to_pandas. Open a JIRA and we can change and make sure it's what we want

Reply:
Done https://issues.apache.org/jira/browse/ARROW-3232